torch/csrc/jit/script/compiler.cpp

```diff
@@ -372,16 +392,26 @@ at::optional<std::vector<Value*>> tryMatchSchema(
 
     std::vector<at::optional<NamedValue>> positional_inputs(schema.arguments.size(), at::nullopt);
 
-    size_t total_inputs = attributes.size() + inputs.size();
-    if(total_inputs > schema.arguments.size()) {
-      err() << "expected at most " << schema.arguments.size() << " arguments "
-      << " but found " << total_inputs << "\n" << loc << "\n";
-      return at::nullopt;
-    }
     // fill in positional arguments
-    for(size_t i = 0; i < inputs.size(); ++i) {
+    size_t non_vararg = std::min(inputs.size(), schema.arguments.size());
+    for(size_t i = 0; i < non_vararg; ++i) {
       positional_inputs[i] = inputs[i];
     }
+    // if we have extra vararg arguments pack them all up into a tuple as the last argument
```

Comment:
Tbh this logic is a bit weird for me. In a call like `expand(tensor, 1, 2, 3, 4)`, this will initially fill in `positional_inputs` with `{tensor, 1}`. It would be clearer if we did a check for vararg function before, and took different paths based on that.

Another weird thing with this code (that just so happens to work because we represent singleton tuples in the same way as elements) is that a single argument doesn't get promoted to a tuple type. This is a recipe for a breakage in the future.